container_opts.go

```diff
@@ -255,9 +223,19 @@ func WithSnapshotCleanup(ctx context.Context, client *Client, c containers.Conta
 	return nil
 }
 
+// WithNewSnapshot allocates a new snapshot to be used by the container as the
+// root filesystem in read-write mode
+func WithNewSnapshot(id string, i Image, opts ...snapshots.Opt) NewContainerOpts {
+	return withNewSnapshotBase(id, i, false, opts...)
+}
+
 // WithNewSnapshotView allocates a new snapshot to be used by the container as the
 // root filesystem in read-only mode
 func WithNewSnapshotView(id string, i Image, opts ...snapshots.Opt) NewContainerOpts {
+	return withNewSnapshotBase(id, i, true, opts...)
+}
+
+func withNewSnapshotBase(id string, i Image, readonly bool, opts ...snapshots.Opt) NewContainerOpts {
```

Comment:
nit: 
```suggestion
func withNewSnapshotBase(id string, i Image, isView bool, opts ...snapshots.Opt) NewContainerOpts {
```

`isView` might be clearer name.

Reply:
Thanks for the suggestion. I used `readonly` to keep it consistent with other similar usages (the PR description has 3 other places where `readonly` is used to decide to call `View` or `Prepare`). Maybe we can keep the consistency here since it might be more familiar to people. I can also update all appearance to `isView` if preferred. :)

Reply:
> Thanks for the suggestion. I used readonly to keep it consistent with other similar usages (the PR description has 3 other places where readonly is used to decide to call View or Prepare). Maybe we can keep the consistency here since it might be more familiar to people. I can also update all appearance to isView if preferred. :)

@djdongjin Thanks for the explanation. Then `readonly` sounds good to me :smile: